Assets/_BForBoss/_Core/Scripts/Boss/DerekBossManager.cs

```diff
@@ -1,24 +1,50 @@
 using System;
+using Perigon.Utility;
+using Perigon.Weapons;
 using UnityEngine;
 
 namespace BForBoss
 {
     [DisallowMultipleComponent]
-    public class DerekBossManager : MonoBehaviour
+    [RequireComponent(typeof(Collider))]
+    public class DerekBossManager : MonoBehaviour, IBulletCollision
     {
-        private Action _onPhaseComplete;
+        [SerializeField, Resolve] private DerekShieldBehaviour _shieldBehaviour;
+        [SerializeField, Resolve] private BossWipeOutWallsManager _wipeoutWallsManager;
+        [SerializeField] private DerekMissileLauncherBehaviour[] _missileLauncherBehaviours;
+        [SerializeField, Tooltip("Temporary timer to space out Missile shots for testing purposes"), Min(0.0f)] private float _timeBetweenMissileShots = 5.0f;
+        [SerializeField, Tooltip("Temporary timer to track how long Derek has been in its vulnerable state"), Min(0.0f)] private float _vulnerabilityDuration = 10.0f; 
         
-        public void Initialize(Action onPhaseComplete)
-        {
-            _onPhaseComplete = onPhaseComplete;
-        }
+
+        private Action<bool> _onVulnerabilityExpired;
+        private bool _isVulnerable = false;
+        private float _vulnerabilityTimer;
         
         public void Reset()
         {
-            //Reset Derek Position
-            //Reset Derek Death Walls
-            //Reset Missile Behavior to default Params
-            //Raise Shields
+            _isVulnerable = false;
+            _wipeoutWallsManager.Reset();
+            foreach (DerekMissileLauncherBehaviour missileLauncher in _missileLauncherBehaviours)
+            {
+                missileLauncher.Reset();
+            }
+            _shieldBehaviour.ToggleShield(false);
+        }
+
+        public void Initialize(PlayerMovementBehaviour playerMovementBehaviour, Action<bool> onVulnerabilityExpired)
+        {
+            _wipeoutWallsManager.Initialize(playerMovementBehaviour);
+            foreach (DerekMissileLauncherBehaviour missileLauncher in _missileLauncherBehaviours)
+            {
+                if (missileLauncher == null)
```

Comment:
When would missileLauncher ever be null? 
If it's because of a design mistake, then you should check if it's null onawake and PanicHelper crash the game. Instead of needing to check each time you loop through.